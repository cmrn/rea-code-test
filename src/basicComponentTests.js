import React from 'react';
import { shallow } from 'enzyme';

export default function basicComponentTests(component) {
  it('allows a className to be set', () => {
    const wrapper = shallow(React.cloneElement(component, { className: "classy" }));
    expect(wrapper.props().className).toEqual(expect.stringContaining('classy'));
  });

  it('passes through other props', () => {
    const wrapper = shallow(React.cloneElement(component, { foo: "bar" }));
    expect(wrapper.props().foo).toEqual('bar');
  });
}
