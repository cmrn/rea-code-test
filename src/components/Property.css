/* Material card & hover effect from https://codepen.io/sdthornton/pen/wBZdXq */

.Property {
  margin: 20px 0;
  width: 320px;
  
  background: #fff;
  border-radius: 2px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
}

.Property:hover {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
  cursor: pointer;
}

.Property-image {
  width: 100%;
}

.Property-price {
  font-size: 26px;
  text-align: right;
  padding: 10px 20px;
}