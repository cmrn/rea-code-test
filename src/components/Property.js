import React from 'react';
import cnames from 'classnames';
import './Property.css';

const Property = ({property, className, ...props}) => (
  <div className={cnames('Property', className)} {...props}>
    <img className='Property-image' src={property.mainImage} />
    <div className='Property-price'>{property.price}</div>
  </div>
);

export default Property;
