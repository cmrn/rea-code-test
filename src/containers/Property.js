import React from 'react';
import { view } from 'react-easy-state';
import Property from '../components/Property';
import propertyStore from '../propertyStore';

const PropertyContainer = ({id, ...props}) => (
  <Property 
    {...props} 
    property={propertyStore.properties[id]}
  />
);

export default view(PropertyContainer);
