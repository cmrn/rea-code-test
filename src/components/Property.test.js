import React from 'react';
import basicComponentTests from '../basicComponentTests';
import { shallow } from 'enzyme';
import Property from './Property';

const property = {
  "price": "$726,500",
  "agency": {
      "brandingColors": {
          "primary": "#ffe512"
      },
      "logo": "http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
  },
  "id": "1",
  "mainImage": "http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
};

basicComponentTests(<Property property={property} />);

it('renders the property with image, agency, and price', () => {
  const wrapper = shallow(<Property property={property} />);
  expect(wrapper).toMatchSnapshot();
});