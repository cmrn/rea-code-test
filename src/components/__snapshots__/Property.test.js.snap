// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the property with image, agency, and price 1`] = `
<div
  className="Property"
>
  <AgencyBanner
    color="#ffe512"
    logo="http://i1.au.reastatic.net/agencylogo/XRWXMT/12/20120927204448.gif"
  />
  <img
    className="Property-image"
    src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
  />
  <div
    className="Property-price"
  >
    $726,500
  </div>
</div>
`;
